packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -42,4 +50,63 @@ class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJa
       }
     }
   }
+
+  @ReactMethod
+  fun queryCache(urls: ReadableArray, promise: Promise) {
+    moduleCoroutineScope.launch {
+      try {
+        val resultMap = WritableNativeMap()
+        urls.toArrayList()
+            .filterIsInstance<String>()
+            .map { url -> async { Triple(url, isInCache(url, ImageCacheType.DISK), isInCache(url, ImageCacheType.MEMORY))  } }
+            .awaitAll()
+            .filter {
+              if (it.second == null || it.third == null) {
+                throw InterruptedException()
+              } else {
+                it.second == true || it.third == true
+              }
+            }
```

Comment:
```suggestion
            .filter { (_, inDisk, inMemory) -> inDisk || inMemory }
```
See the comment below about exceptions